### ISSUE TITLE
[Enhancement]: aws_elasticache_replication_group resource for REDIS does NOT allow to provide 7.x the same way it allows us to use 6.x

### ISSUE THREAD

**Issue by 1968mnelson** [hashicorp/terraform-provider-aws#30481]
### Description

When REDIS 6 was released. The Hashi provider for AWS allowed us to specify 6.x for the version which would in turn always grab the latest minor and patch versions for v6. Example...using that today would translate to 6.2.6. This makes things very Terraform and pipeline friendly as our customers dont have to care about what the latest minor version is.

Now that REDIS 7 is released....we are being forced to provide the specific minor version for v7...as of today, 7.0.
This is not very pipeline friendly however since have Terraform modules that our customers consume. For example, if they provide 7.0 as the version it will provision that cluster with 7.0...HOWEVER...if auto_minor_version_upgrade is set to True and it upgrades the cluster to the latest during the next maintenance window(7.1 whenever that is actually released), the subsequent deployment in Terraform will detect the drift since TF is providing 7.0 and will replace the cluster to downgrade it back down to 7.0. IF we were able to provide 7.x the same way we do for 6.x, then no drift would be detected on subsequent deploys

I tested the above with 6.0 since 7.0 hasn't actually revved their minor version to 7.1 yet

I DO understand that AWS is the source of this change...we could still provide the redis7 pg family on our end but it would be very helpful if we can provide just 7.x as the version and TF figures out what the latest version from there: 7.0.7, 7.1.0, etc...

### Affected Resource(s) and/or Data Source(s)

aws_elasticache_replication_group 

### Potential Terraform Configuration

```terraform
engine_version = "7.x"
```

### Would you like to implement a fix?

No

**Comment by samyamaiti** [comment]
Getting the below error when used engine_version = "7.x"

`Error creating ElastiCache Replication Group (********): InvalidParameterCombination: Cannot find version 7.x for redis"`

**Comment by 1968mnelson** [comment]
That's correct @samyamaiti ....v7 does not give us the ability to specify 7.x. This is problematic in regards to our TF pipelines... If someone pins to 7.0 BUT inadvertently sets their auto_upgrade_minor_version to true....then any subsequent deployments with the same TF code would detect a downgrade and flag the cluster for replacement.
It appears that this is on the AWS side as I do NOT see a corresponding 7.x parameter group either in AWS...its just redis7
My hope is that Hashicorp provider can maybe abstract that away from us and determine the latest minor.patch if we provide 7.x for the version

**Comment by 1968mnelson** [comment]
Quick update on this. I did speak with AWS support and they have also created an internal ticket to allow 7.x version and parameter group. They said it will most likely come whenever their first 7.1 minor version is released.
I will update here accordingly
